src/status_im/contexts/onboarding/create_password/view.cljs

```diff
@@ -29,78 +27,91 @@
     (i18n/label :t/password-creation-subtitle)]])
 
 (defn password-inputs
-  [{:keys [passwords-match? on-change-password on-change-repeat-password on-input-focus
-           password-long-enough? password-short-enough? empty-password? show-password-validation?
-           on-blur-repeat-password]}]
-  (let [hint-1-status (if password-long-enough? :success :default)
-        hint-2-status (if passwords-match? :success :error)
-        hint-2-text   (if passwords-match?
-                        (i18n/label :t/password-creation-match)
-                        (i18n/label :t/password-creation-dont-match))
-        error?        (and show-password-validation?
-                           (not passwords-match?)
-                           (not empty-password?))]
+  [{:keys [set-password set-repeat-password same-password-length? same-passwords?
+           password-long-enough? password-short-enough? non-empty-password?]}]
+  (let [[show-validation?
+         set-show-validation?]    (rn/use-state false)
+        on-change-password        (rn/use-callback
+                                   (fn [new-value]
+                                     (set-password new-value)
+                                     (when same-password-length?
+                                       (set-show-validation? true)))
+                                   [same-password-length?])
+        on-change-repeat-password (rn/use-callback
+                                   (fn [new-value]
+                                     (set-repeat-password new-value)
+                                     (when same-password-length?
+                                       (set-show-validation? true)))
+                                   [same-password-length?])
+        on-blur-repeat-password   (rn/use-callback
+                                   (fn []
+                                     (if non-empty-password?
+                                       (set-show-validation? true)
+                                       (set-show-validation? false)))
+                                   [non-empty-password?])
```

Comment:
These `on-change-*` functions where previously defined in the parent, but they are only used by this child, by moving these now the parent is simpler.